From: Chad Miller <chad.miller@canonical.com>
Subject: defang CT timebomb

Chromium refuses to connect certain https sites (eg. amazon, etsy, ...)
with claiming an insecure connection (CERTIFICATE_TRANSPARENCY_REQUIRED,
BUILD_NOT_TIMELY), which is a false alert, that cannot be bypassed.

The reason behind is a badly implemented new feature called "Certificate
Transparency", which basicly checks some globals logs of collected
site certificates in order to defend against rogue/compromised CAs.
Unfortunately they check the validity of the (built-in) log pointers
by the build-time of the chromium binary. Pretty obvious that this
is just silly - it just makes the whole logic fail miserably, rendering
lots of sites just inaccessible, when the binary isn't recompiled 
regularily.

--- a/net/quic/crypto/proof_verifier_chromium.cc
+++ b/net/quic/crypto/proof_verifier_chromium.cc
@@ -345,6 +345,8 @@ int ProofVerifierChromium::Job::DoVerify
     int ct_result = OK;
     if (verify_details_->ct_verify_result.cert_policy_compliance !=
             ct::CertPolicyCompliance::CERT_POLICY_COMPLIES_VIA_SCTS &&
+        verify_details_->ct_verify_result.cert_policy_compliance !=
+            ct::CertPolicyCompliance::CERT_POLICY_BUILD_NOT_TIMELY &&
         transport_security_state_->ShouldRequireCT(
             hostname_, cert_verify_result.verified_cert.get(),
             cert_verify_result.public_key_hashes)) {
--- a/net/socket/ssl_client_socket_impl.cc
+++ b/net/socket/ssl_client_socket_impl.cc
@@ -1819,6 +1819,8 @@ int SSLClientSocketImpl::VerifyCT() {
 
   if (ct_verify_result_.cert_policy_compliance !=
           ct::CertPolicyCompliance::CERT_POLICY_COMPLIES_VIA_SCTS &&
+      ct_verify_result_.cert_policy_compliance !=
+          ct::CertPolicyCompliance::CERT_POLICY_BUILD_NOT_TIMELY &&
       transport_security_state_->ShouldRequireCT(
           host_and_port_.host(), server_cert_verify_result_.verified_cert.get(),
           server_cert_verify_result_.public_key_hashes)) {
--- a/net/spdy/spdy_session.cc
+++ b/net/spdy/spdy_session.cc
@@ -672,6 +672,8 @@ bool SpdySession::CanPool(TransportSecur
 
   if (ssl_info.ct_cert_policy_compliance !=
           ct::CertPolicyCompliance::CERT_POLICY_COMPLIES_VIA_SCTS &&
+      ssl_info.ct_cert_policy_compliance !=
+          ct::CertPolicyCompliance::CERT_POLICY_BUILD_NOT_TIMELY &&
       transport_security_state->ShouldRequireCT(
           new_hostname, ssl_info.cert.get(), ssl_info.public_key_hashes)) {
     return false;
